data-prepper-plugins/kafka-source/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.source;
+
+import java.util.HashSet;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.Source;
+import org.opensearch.dataprepper.plugins.kafka.source.configuration.TopicsConfig;
+import org.opensearch.dataprepper.plugins.kafka.source.consumer.MultithreadedConsumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.micrometer.core.instrument.Counter;
+
+
+/**
+ * The starting point of the Kafka-source plugin and the Kafka consumer
+ * properties and kafka multithreaded consumers are being handled here.
+ */
+
+@SuppressWarnings("deprecation")
+@DataPrepperPlugin(name = "kafka", pluginType = Source.class, pluginConfigurationType = KafkaSourceConfig.class)
+public class KafkaSource implements Source<Record<Object>> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaSource.class);
+  private final KafkaSourceConfig sourceConfig;
+  private Set<MultithreadedConsumer> consumers = new HashSet<>();
+  private ExecutorService executorService;
+  private static final String KAFKA_WORKER_THREAD_PROCESSING_ERRORS = "kafkaWorkerThreadProcessingErrors";
+  private final Counter kafkaWorkerThreadProcessingErrors;
+  private final PluginMetrics pluginMetrics;
+  private String consumerGroups;
+  private MultithreadedConsumer multithreadedConsumer;
+  private int workers = 0;
+
+  @DataPrepperPluginConstructor
+  public KafkaSource(final KafkaSourceConfig sourceConfig, final PluginMetrics pluginMetrics) {
+    this.sourceConfig = sourceConfig;
+    this.pluginMetrics = pluginMetrics;
+    this.kafkaWorkerThreadProcessingErrors = pluginMetrics.counter(KAFKA_WORKER_THREAD_PROCESSING_ERRORS);
+  }
+  
+  @Override
+  public void start(Buffer<Record<Object>> buffer) {
+    sourceConfig.getTopics().forEach(topic -> {
+      Properties consumerProperties = getConsumerProperties(topic);
+        try {
+        	workers = workers + topic.getTopic().getConsumerGroupConfig().getWorkers(); 
+        	consumerGroups = topic.getTopic().getConsumerGroupConfig().getGroupName();
+            IntStream.range(0, workers).forEach(index -> {
+              String consumerId = Integer.toString(index + 1);
```

Comment:
should we prepend consumerId number with more details like topic+consumerGroup etc? it'll be easier to debug issues related to consumers.